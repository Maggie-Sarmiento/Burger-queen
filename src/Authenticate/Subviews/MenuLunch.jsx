import { Container, Grid, Typography } from '@mui/material';
import React, { useEffect, useState } from 'react';
import CardMenuLunch from '../Components/CardMenuLunch';

const MenuLunch = () => {
  const [dataMenu, setDataMenu] = useState([]);

  useEffect(() => {
    fetch('http://localhost:3001/menuLunch')
      .then((response) => response.json())
      .then((data) => setDataMenu(data));
  }, []);
  return (
    <Container>
      <>
        <Typography
          variant="h4"
          component="h2"
          marginBottom={3}
          marginTop={5}
        >
          Menu Almuerzo y Cena
        </Typography>
        <Grid container spacing={5}>
          {dataMenu.map((product) => (
            <CardMenuLunch product={product} />
          ))}
        </Grid>
      </>
    </Container>
  );
};

export default MenuLunch;
