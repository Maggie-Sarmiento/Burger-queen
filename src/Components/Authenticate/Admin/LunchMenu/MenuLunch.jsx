/* eslint-disable no-underscore-dangle */
/* eslint-disable react/prop-types */
import { Container, Grid, Typography } from '@mui/material';
import React, { useEffect, useState } from 'react';
import Button from '@mui/material/Button';
import AddIcon from '@mui/icons-material/Add';
import CardMenu from '../Menu/CardMenu/CardMenu';
import ModalMenu from '../Menu/ModalMenu/ModalMenu';
import style from '../Menu/Menu.module.css';
import { urlServer } from '../../../../conf';

const btnStyle = {
  bgcolor: 'black',
  color: '#F3B240',
  fontWeight: 600,
  marginTop: '1em',
  marginRight: '50em',
  width: '10em',
};

const MenuLunch = ({ role }) => {
  const [dataMenu, setDataMenu] = useState([]);
  const [refreshData, setRefreshData] = useState(false);
  const [open, setOpen] = useState(false);
  const handleOpen = () => setOpen(true);

  useEffect(() => {
    const requestOption = {
      method: 'GET',
      headers: { 'Content-Type': 'application/json', Authorization: localStorage.getItem('token') },
    };
    fetch(`${urlServer}/menuLunch`, requestOption)
      .then((response) => response.json())
      .then((data) => setDataMenu(data));
  }, [refreshData]);

  return (
    <section className={style.sectionMenu}>
      {(role === 'Admin')
        && (
        <div className={style.btnModal}>
          <Button sx={btnStyle} onClick={handleOpen}>
            <AddIcon />
            Producto
          </Button>
        </div>
        )}
      <ModalMenu
        menu="menuLunch"
        open={open}
        setOpen={setOpen}
        refreshData={refreshData}
        setRefreshData={setRefreshData}
      />
      <Container>
        <>
          <Typography
            variant="h4"
            component="h2"
            marginBottom={3}
            marginTop={5}
          >
            Menu Almuerzo y Cena
          </Typography>
          <Grid container spacing={3}>
            {dataMenu.map((product) => (
              <CardMenu
                key={product._id}
                role={role}
                menu="menuLunch"
                product={product}
                refreshData={refreshData}
                setRefreshData={setRefreshData}
              />
            ))}
          </Grid>
        </>
      </Container>
    </section>
  );
};

export default MenuLunch;
