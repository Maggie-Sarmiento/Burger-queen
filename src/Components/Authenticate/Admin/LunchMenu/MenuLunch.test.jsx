/* eslint-disable no-undef */
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { create, act } from 'react-test-renderer';
import MenuLunch from './MenuLunch';

describe('test header', () => {
  test('render header', async () => {
    let root;
    const mockRole = jest.fn();
    act(() => {
      root = create(
        <Router>
          <MenuLunch role={mockRole} />
        </Router>,
      );
    });
    expect(root.toJSON()).toMatchSnapshot();
  });
});
