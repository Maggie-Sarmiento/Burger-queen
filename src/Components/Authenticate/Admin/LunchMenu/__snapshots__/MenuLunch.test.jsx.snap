// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test header render header 1`] = `
<section
  className="sectionMenu"
>
  <section />
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <h2
      className="MuiTypography-root MuiTypography-h4 css-10mgom-MuiTypography-root"
    >
      Menu Almuerzo y Cena
    </h2>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
    />
  </div>
</section>
`;
